Source/Libraries/CorruptCore/Stockpile/Stockpile.cs

```diff
@@ -23,14 +23,12 @@ public class Stockpile
         [Exclude]
         private static NLog.Logger logger = NLog.LogManager.GetCurrentClassLogger();
 
-        public List<StashKey> StashKeys = new List<StashKey>();
+        public List<StashKey> StashKeys { get; private set; } = new List<StashKey>();
 
-        public string Name;
-        public string Filename;
-        public string ShortFilename;
-        public string RtcVersion;
-        public string VanguardImplementation;
-        public bool MissingLimiter;
+        public string Filename { get; private set; }
```

Comment:
I'm assuming `public`/`private` isn't relevant for serialization?

Reply:
Now I'm not removing any fields, but I've made many of them private. Also noteworthy that `Filename` and `MissingLimit` are now properties instead of fields.

Reply:
Just to be safe, I reverted all of the field -> property changes.

Reply:
I _think_ the serialization is configured to serialize all fields by default and not just public, but that's probably worth double-checking

Reply:
> Although it is not explicitly demonstrated in this example, **all member variables of a class will be serialized—even variables marked as private.**

https://docs.microsoft.com/en-us/dotnet/standard/serialization/basic-serialization

Reply:
Json.Net and Ceras abide by their own rules. 

In this case, it's being saved to a stockpile so we're dealing with Json.Net which'll only serialize public fields by default.
https://www.newtonsoft.com/json/help/html/SerializationGuide.htm

Checking JsonHelper, it's not set to serialize all fields so if we're reducing visibility, the fields have to be tagged with  [JsonPropertyAttribute]

Reply:
Done

Reply:
Sounds good. We can take a stab at these in a separate commit where it's easier to test for any accidental regressions since it's not lumped with 50 other changes.

Reply:
Got it. I missed a bunch of serializable classes and I'm cleaning that up now. I'll un-draft the PR when I'm done. The code is a bit messy w/ suppressions but I can clean them up in a separate change

Reply:
Alright, I'm pretty sure I caught all of the serializable classes where I made changes.